File: guides/micronaut-jaxrs-jdbc/java/src/main/java/example/micronaut/domain/Pet.java
package example.micronaut.domain;

import io.micronaut.core.annotation.Creator;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.data.annotation.AutoPopulated;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

import java.util.UUID;

@MappedEntity
public class Pet {

    @Id
    @AutoPopulated
    private UUID id;

    private String name;

    //Mapped with Micronaut Data
    //@ManyToOne
    private Owner owner;

    private PetType type = PetType.DOG;

    @Creator
    public Pet(String name, @Nullable Owner owner) {
        this.name = name;
        this.owner = owner;
    }

    public Owner getOwner() {
        return owner;
    }

    public String getName() {
        return name;
    }

    public UUID getId() {
        return id;
    }

    public PetType getType() {
		return type;
	}

	public void setType(PetType type) {
		this.type = type;
	}

	public void setId(UUID id) {
        this.id = id;
    }

}

File: guides/micronaut-jaxrs-jdbc/python/src/example/micronaut/pet.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.validation.constraints import NotBlank, NotNull
from micronaut.data.annotation import GeneratedValue, Id, MappedEntity
from micronaut.serde.annotation import Serdeable

from .pet_type import PetType


@dataclass
@Serdeable  # <1>
@MappedEntity  # <2>
class Pet:
    name: Annotated[str, NotBlank]
    type: Annotated[PetType, NotNull] = PetType.DOG
    id: Annotated[int | None, Id, GeneratedValue] = None  # <3> <4>
